test/e2e/runner/rpc.go

```diff
@@ -182,3 +195,35 @@ func waitForNode(ctx context.Context, node *e2e.Node, height int64) (*rpctypes.R
 		}
 	}
 }
+
+// getLatestBlock returns the last block that all active nodes in the network have
+// agreed upon i.e. the earlist of each nodes latest block
+func getLatestBlock(ctx context.Context, testnet *e2e.Testnet) (*types.Block, error) {
+	var earliestBlock *types.Block
+	for _, node := range testnet.Nodes {
+		// skip nodes that don't have state or haven't started yet
+		if node.Stateless() {
+			continue
+		}
+		if !node.HasStarted {
+			continue
+		}
```

Comment:
Maybe:

```suggestion
		if node.Stateless() || !node.HasStarted {
			continue
		}
```

Reply:
I put them in different statements just because it's easier to read.

Reply:
No objections. I was mostly just making the code echo the language of the comment. 🙂